CSharpFunctionalExtensions/Maybe/Maybe.cs

```diff
@@ -142,23 +142,23 @@ public override bool Equals(object obj)
             return false;
         }
 
-        public bool Equals(Maybe<T> other)
+        public bool Equals(Maybe<T> other, IEqualityComparer<T> comparer = null)
         {
             if (HasNoValue && other.HasNoValue)
                 return true;
 
             if (HasNoValue || other.HasNoValue)
                 return false;
 
-            return EqualityComparer<T>.Default.Equals(_value, other._value);
+            return comparer.Equals(_value, other._value) ?? EqualityComparer<T>.Default.Equals(_value, other._value);
```

Comment:
`comparer.Equals` throws if `comparer` is null. The same applies for `comparer.GetHashCode`